### PR TITLE
re-use workers for multiple tasks

### PR DESCRIPTION
**Pull request by kleinschmidt**
the issue was that we were not setting `runtime_env_hash` correctly in the core worker options.

~~the test needs work: currently, it'll spin up more workers to handle the extra load.  informally when I tested htis locally, running the test twice, it does re-use 8 workers (number of CPUs on my macbook air).~~

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205303690266614

### CODE REVIEW

File: deps/wrapper.cc [hunk 1]
```diff
@@ -79,6 +80,7 @@ void initialize_worker(
     options.raylet_ip_address = node_ip_address;
     options.metrics_agent_port = -1;
     options.startup_token = startup_token;
+    options.runtime_env_hash = runtime_env_hash;
```
Comment (glennmoy) [inline deps/wrapper.cc:83]: huh I actually came across this last week and tried harcoding `runtime_env_hash` but found it still didn't work

https://github.com/beacon-biosignals/ray/blob/4ceb62daaad05124713ff9d94ffbdad35ee19f86/src/ray/raylet/worker.h#L274-L277

**Review by glennmoy — APPROVED** [review]
LGTM pending an appropriate test

File: deps/wrapper.cc (hunk 1 above)
Comment (glennmoy) [reply deps/wrapper.cc:83]: ah this actually corroborates what you saw in the logs about it not being properly propagated! `runtime_env_hash: 1462190447 cached runtime env hash: 0`

Comment (kleinschmidt) [reply deps/wrapper.cc:83]: ah you were on teh right track long before I was!

File: Ray.jl/test/task.jl [hunk 2]
```diff
@@ -151,19 +151,10 @@ end
...
-    end
-
-    @time pids = Ray.get.(oids)
-    @test !allunique(pids)
+    n_tasks = Sys.CPU_THREADS * 2
+    # warm up worker cache pool
+    pids = Ray.get.([submit_task(getpid, ()) for _ in 1:n_tasks])
+    # run more tasks which should re-use the cpu cache
+    pids2 = Ray.get.([submit_task(getpid, ()) for _ in 1:n_tasks])
+    @test !isempty(intersect(pids, pids2))
```
Comment (glennmoy) [inline Ray.jl/test/task.jl:159]: test is effectively saying "at least one process should get re-used" which is probably good enough to show it works

**Review by glennmoy — APPROVED** [review]

File: Ray.jl/test/task.jl [hunk 3]
```diff
@@ -149,3 +149,12 @@ end
         end
     end
 end
+
+@testset "many tasks" begin
+    n_tasks = Sys.CPU_THREADS * 2
```
Comment (kleinschmidt) [inline Ray.jl/test/task.jl:154]: we can  maybe get away with smaller number of tasks but want to make sure we saturate teh cache of workers

**Pull request merged**